### PR TITLE
Examples for creating Custom ColorCycles and ColorMaps

### LINKED ISSUE #24485: Create and set custom colormap in matplotlib
**Original reporter:** Nina, ISIS

The reflectometers need a way to set a custom colormap and custom color cycle for the line plots. They are happy with this to be scripted so this is most likely a documentation issue where we create further examples in the [documentation](http://docs.mantidproject.org/nightly/plotting/index.html#plotting) for users to follow.

### PR DESCRIPTION
**Pull request by DanielMurphy22**
**Description of work**

Writing some example scripts and instructions for plotting with a Custom ColorCycle and creating Custom Colormaps.

Fixes #24485 
refs #27503

**To test:**

Build docs.html, check the changes look correct and the examples work!

*This does not require release notes* because **it's Documentation**
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by cailafinn — CHANGES_REQUESTED** [review]
Looks good and the script examples run well. Just a few things to look at.

File: docs/source/plotting/index.rst [hunk 1]
```diff
@@ -402,10 +404,303 @@ One can do twin axes as well:
    fig.tight_layout()
    #fig.show()
 
+Custom Colors
+-------------
+
+Custom Color Cycle (Line / 1D plots)
+####################################
+
+The Default Color Cycle doesn't have to be used. Here is an example where a Custom Color Cycle is chosen. Make sure to fill the list `custom_colors` with the HTML codes for the desired colours.
```
[the remark this answers is not in the document]
Comment (DanielMurphy22) [reply docs/source/plotting/index.rst]: Cannot be rbg or rgba, must be html code or the name of a color on the html code list. Will Make all changes on Monday!

File: docs/source/plotting/index.rst [hunk 2]
```diff
@@ -410,24 +410,25 @@ Custom Colors
 Custom Color Cycle (Line / 1D plots)
 ####################################
 
-The Default Color Cycle doesn't have to be used. Here is an example where a Custom Color Cycle is chosen. Make sure to fill the list `custom_colors` with the HTML codes for the desired colours. 
-These can be found `online <https://www.rapidtables.com/web/color/html-color-codes.html>`_ or using Mantid Workbench: On Workbench, plot a Spectrum and navigate to Figure Options(Gear Icon) > Curves > Line > Color (RGBA). Click on the nearby colorbox and use the 'Select Color' pop-up to chose your colors and their associated HTML codes. `Further Details <https://matplotlib.org/3.1.1/gallery/color/color_cycle_default.html>`_
+The Default Color Cycle doesn't have to be used. Here is an example where a Custom Color Cycle is chosen. Make sure to fill the list `custom_colors` with either the HTML codes or recognised names for the desired colours.
```
Comment (ConorMFinn) [inline docs/source/plotting/index.rst]: > HTML codes

I still don't think this is clear enough as HTML can support a range of formats for colour. If the colour code has to be in the form `#b3457f` then you should specify "HTML _hex_ code".

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just one thing to comment on with the main document. 

I feel I should also ask if the images are still being used, you seem to generate them using the documentation at the moment.

File: docs/source/plotting/index.rst (hunk 2 above)
Comment (DanielMurphy22) [reply docs/source/plotting/index.rst]: Latest commit adds in hex
AND removes the ColorCycle Image as you are correct I'm now Plotting this.
The ColorMapCustomPlot.PNG is from MantidPlot, not using Matplolib so needs to remain as an image!

**Review by cailafinn — APPROVED** [review]
This looks good to me now. Approved :+1:

**Pull request merged**